python/instrumentation/openinference-instrumentation-llama-index/src/openinference/instrumentation/llama_index/__init__.py

```diff
@@ -30,13 +32,44 @@ def _instrument(self, **kwargs: Any) -> None:
             OpenInferenceTraceCallbackHandler,
         )
 
-        import llama_index.core
+        if _legacy_llama_index():
```

Comment:
I think we could just publish with a `use_callback_handler` flag. That way we could get this in more hands early?

Reply:
OK. I'll add a flag that _adds_ instrumentation on top of the old paradigm so they're both exported at the same time for ease of comparison.